tools/dotnet-linker/Steps/GenerateReferencesStep.cs

```diff
@@ -0,0 +1,49 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+
+using Mono.Cecil;
+
+using Xamarin.Bundler;
+using Xamarin.Linker;
+
+namespace Xamarin {
+
+	public class GenerateReferencesStep : ConfigurationAwareStep {
+		protected override string Name { get; } = "Generate References";
+		protected override int ErrorCode { get; } = 2320;
+
+		protected override void TryEndProcess ()
+		{
+			base.TryEndProcess ();
+
+			var app = Configuration.Application;
+			var required_symbols = Configuration.DerivedLinkContext.RequiredSymbols;
+			var items = new List<MSBuildItem> ();
+				
+			switch (app.SymbolMode) {
+			case SymbolMode.Ignore:
+				break;
+			case SymbolMode.Code:
+				string reference_m = Path.Combine (Configuration.CacheDirectory, "reference.m");
+				reference_m = Configuration.Target.GenerateReferencingSource (reference_m, required_symbols);
+				if (!string.IsNullOrEmpty (reference_m)) {
+					var item = new MSBuildItem { Include = reference_m };
+					items.Add (item);
+				}
+				Configuration.WriteOutputForMSBuild ("_ReferencesFile", items);
+				break;
+			case SymbolMode.Linker:
+			case SymbolMode.Default:
+				foreach (var symbol in required_symbols) {
+					var item = new MSBuildItem { Include = "-u" + symbol.Prefix + symbol.Name };
```

Comment:
I think you need a space here:

```suggestion
					var item = new MSBuildItem { Include = "-u " + symbol.Prefix + symbol.Name };
```

although that might have to be treated as two different MSBuildItems.

Reply:
I actually checked it and the space is not necessary. In fact the space would create a problem because it would pass the whole thing quoted to the linker and the linker would expect the space to be part of the symbol name.